plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -76,13 +78,13 @@
 
   private static AnalyticsPingManager instance;
 
+  private final String endpointUrl;
   // Preference store (should be configuration scoped) from which we get UUID, opt-in status, etc.
-  private IEclipsePreferences preferences;
-  private Display display;
-  private boolean analyticsEnabled;
+  private final IEclipsePreferences preferences;
+  private final Display display;
 
-  private ConcurrentLinkedQueue<PingEvent> pingEventQueue;
-  private Job eventFlushJob = new Job("Analytics Event Submission") {
+  private final ConcurrentLinkedQueue<PingEvent> pingEventQueue;
```

Comment:
Could be declared as `Queue`, but if we insist in making it a concurrent queue, we could use `ConcurrentArrayQueue` instead, whose javadoc says

>This class is a drop-in replacement for ConcurrentLinkedQueue, with similar performance but producing less garbage because arrays are used to store elements rather than nodes.